src/lib/components/shared/text_editor.svelte

```diff
@@ -134,7 +138,12 @@
                 element.value = element.value.substring(0, selection_start - 1) + replacement_text + element.value.substring(selection_end + replacement_text.length + 2);
             } else {
                 const replacement_text = `_${selection_text}_`;
-                element.value = element.value.substring(0, selection_start) + replacement_text + element.value.substring(selection_end + replacement_text.length);
+                element.value = element.value.substring(0, selection_start) + replacement_text + element.value.substring(selection_end);
+                // reset caret position
+                tick().then(() => {
+                    const caret_position = element.value.substring(0, selection_start).length + replacement_text.length - 1;
+                    element.setSelectionRange(caret_position, caret_position);
+                });
```

Comment:
This functionality is not good imho. If you want to improve, do what github is doing.

Reply:
> This functionality is not good imho. If you want to improve, do what github is doing.

I think it its better than before ( before: moving caret at the end of value instead newly updated text )

Reply:
> I think it its better than before ( before: moving caret at the end of value instead newly updated text )

But then again whats the point? If we want to do something like this, it would be better if we are doing what github is doing. EG: Keep the text selected.

Reply:
> it would be better if we are doing what github is doing. EG: Keep the text selected.

Yea right, check now